plugins/dmxcontrol3/ddf3-functions.js

```diff
@@ -0,0 +1,908 @@
+const xmlbuilder = require(`xmlbuilder`);
+
+const { Capability, CoarseChannel } = require(`../../lib/model.js`);
+
+module.exports = {
+  const: {
+    isCapSuitable: cap => cap._channel.isConstant,
+    create: (channel, caps) => {
+      const xmlConst = xmlbuilder.create(`const`);
+      xmlConst.attribute(`val`, channel.getDefaultValueWithResolution(CoarseChannel.RESOLUTION_8BIT));
+      return xmlConst;
+    }
+  },
+  dimmer: {
+    isCapSuitable: cap => cap.type === `Intensity`,
+    create: (channel, caps) => {
+      const xmlDimmer = xmlbuilder.create(`dimmer`);
+
+      if (channel.capabilities.length > 1 || caps[0].brightness[0].number !== 0) {
+        const dmxControlCaps = getSingleUnitCapabilities(caps, `brightness`, `%`);
+        dmxControlCaps.forEach(cap => {
+          const xmlCap = getBaseXmlCapability(cap.capObject, cap.startValue, cap.endValue);
+          xmlCap.attribute(`type`, `linear`);
+          xmlDimmer.importDocument(xmlCap);
+        });
+      }
+
+      return xmlDimmer;
+    }
+  },
+  shutter: {
+    isCapSuitable: cap => {
+      const isShutterCap = cap.type === `ShutterStrobe` && [`Open`, `Closed`].includes(cap.shutterEffect);
+      const channelHasOpen = cap._channel.capabilities.some(cap => cap.shutterEffect === `Open`);
+      const channelHasClosed = cap._channel.capabilities.some(cap => cap.shutterEffect === `Closed`);
+      return isShutterCap && channelHasOpen && channelHasClosed;
+    },
+    create: (channel, caps) => {
+      const xmlShutter = xmlbuilder.create(`shutter`);
+
+      caps.forEach(cap => {
+        const xmlCap = getBaseXmlCapability(cap);
+        xmlCap.attribute(`type`, cap.shutterEffect.toLowerCase());
+        xmlShutter.importDocument(xmlCap);
+      });
+
+      return xmlShutter;
+    }
+  },
+  strobe: {
+    isCapSuitable: cap =>
+      (cap.type === `ShutterStrobe` && ![`Open`, `Closed`].includes(cap.shutterEffect)) ||
+      (cap.type === `NoFunction` && cap._channel.type === `Strobe`),
+    create: (channel, caps) => {
+      const xmlStrobe = xmlbuilder.create(`strobe`);
+
+      caps.forEach(cap => {
+        let xmlCap;
+
+        if (cap.speed) {
+          const dmxControlCap = getSingleUnitCapabilities([cap], `speed`, `Hz`, 0, 50)[0];
```

Comment:
Maybe this is a bit easier to read:

```suggestion
          const [dmxControlCap] = getSingleUnitCapabilities([cap], `speed`, `Hz`, 0, 50);
```